training_record/services.py

```diff
@@ -141,6 +143,76 @@ def create_campus_records_from_excel(file):
 
         return len(records)
 
+    @staticmethod
+    def department_admin_review(record_id, is_approved, user):
+        '''Department admin review the off-campus training record.'''
+        record = Record.objects.filter(pk=record_id, campus_event__isnull=True)
```

Comment:
Move these lines into the scope of `with transaction.atomic()`, and add `.select_for_update()` to lock the record.